rewiringamerica/embed.rewiringamerica.org
# Add email signup form to empty state

## Description

If the `show-email` attribute is present on the calculator, the empty
state of the incentive results grid will show an extra thing below the
copy. Depending on whether the value in local storage indicates you've
already signedup, it's either "you're subscribed to our newsletter!"
or a CTA to subscribe. Upon submitting the form, the UI immediately
changes to the already-subscribed state.

If `show-email` isn't present, everything is as before.

One intentional difference from the design spec is that the button
text in the form is "Subscribe" (the spec says "Submit"). More
specific action verbs are preferable, generally.

TODO:

- [x] Spanish translations
- [x] Remove `show-email`; it's just there for the preview

## Test Plan

Use the inputs 02859 / "cooking" only / $200,000 / 1 person / single
to get an empty state.

- With `show-email` and no signup key in local storage, calculate and
  make sure the empty state looks right (on all 3 layouts), and the
  "back to calculator" pseudo-link still works.

- Submit the email form and make sure the UI changes to
  already-subscribed state. Make sure the email field is also gone
  from the main form.

- Delete the local storage key, then submit the main form with an
  email included. Make sure the already-subscribed state shows up
  below.

- Without `show-email`, make sure there's nothing about subscribing in
  the empty state, regardless of local storage key.
